index.js

```diff
@@ -38,21 +38,23 @@ const {UDP_KEY} = require('./lib/config');
  */
 class TuyaDevice extends EventEmitter {
   constructor({
-    ip,
-    port = 6668,
-    id,
-    gwID = id,
-    key,
-    productKey,
-    version = 3.1,
-    nullPayloadOnJSONError = false,
-    issueGetOnConnect = true
-  } = {}) {
+                ip,
+                port = 6668,
+                id,
+                gwID = id,
+                key,
+                productKey,
+                version = 3.1,
+                nullPayloadOnJSONError = false,
+                issueGetOnConnect = true,
+                issueRefreshOnConnect = false
+              } = {}) {
     super();
     // Set device to user-passed options
     this.device = {ip, port, id, gwID, key, productKey, version};
     this.globalOptions = {
-      issueGetOnConnect
+      issueGetOnConnect,
+      issueRefreshOnConnect
```

Comment:
do we need to add checks if someone provides both? or should refresh noch be "the fallback" if get do not work?

Reply:
make sense, no idea yet, how to do it :D